indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskLockbox.java

```diff
@@ -1271,6 +1278,138 @@ public void remove(final Task task)
     }
   }
 
+  /**
+   * Acquire a read lock to perform the segment transactional append action for a given datasource.
+   * Also verifies that all the locks are of the type APPEND for the task.
+   * @param task task to perform the append action
+   */
+  public void acquireTransactionalAppendLock(Task task)
+  {
+    acquireTransactionalAppendLock(task, LOCK_ACQUIRE_TIMEOUT_MILLIS);
+  }
+
+  @VisibleForTesting
+  void acquireTransactionalAppendLock(Task task, long lockAcquireTimeoutMillis)
```

Comment:
Why does this need to be exposed? You can have a test that waits for 10s.

Reply:
Made this public instead